packages/openapi3/src/xml-module.ts

```diff
@@ -122,19 +121,12 @@ export async function resolveXmlModule(): Promise<XmlModule | undefined> {
         const propXmlName = hasUnwrappedDecorator
           ? xmlName
           : resolveEncodedName(program, propValue as Scalar | Model, "application/xml");
-        if (propValue.kind === "Scalar") {
-          let scalarSchema: OpenAPI3Schema = {};
-          const isStd = program.checker.isStdType(propValue);
-          if (isStd) {
-            scalarSchema = getSchemaForStdScalars(propValue, options);
-          } else if (propValue.baseScalar) {
-            scalarSchema = getSchemaForStdScalars(
-              propValue.baseScalar as Scalar & { name: IntrinsicScalarName },
-              options,
-            );
-          }
-          scalarSchema.xml = { name: propXmlName };
-          refSchema.items = scalarSchema;
+
+        if ("type" in refSchema.items) {
```

Comment:
why check "type" here? what about [playground](https://typespec.io/playground/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwpAc2VydmljZSgjeyB0aXRsZTogIldpZGdldCBTxhoiIH0pCm5hbWVzcGFjZSBEZW1vxxo7CgpAZG9jKFQpCnNjYWxhciB4PFQgZXh0ZW5kcyB2YWx1ZW9mIHN0cmluZz47Cgptb2RlbCB5IHsKICBhOiB4PCJhYWEiPltdCn0%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D)

Reply:
do you want to check if it is inline or not like this: https://github.com/microsoft/typespec/blob/main/packages/openapi3/src/schema-emitter-3-0.ts#L206?